pkg/virt-controller/watch/vmi/vmi_test.go

```diff
@@ -245,6 +240,14 @@ var _ = Describe("VirtualMachineInstance watcher", func() {
 		rqInformer, _ := testutils.NewFakeInformerFor(&k8sv1.ResourceQuota{})
 		nsInformer, _ := testutils.NewFakeInformerFor(&k8sv1.Namespace{})
 		var qemuGid int64 = 107
+
+		netStatusUpdated = false
+
+		stubNetStatusUpdate := func(_ *virtv1.VirtualMachineInstance, _ *k8sv1.Pod) error {
+			netStatusUpdated = true
+			return nil
+		}
```

Comment:
I think you can avoid the nasty variable by just mutating the VMI. Perhaps add an annotation and validate it is there. It can be a simple function with some text explaining what it is simulating.

Reply:
I've tried mutating the VMI, since the code works on a deep copy - the change is not reflected at the test's level.

Reply:
It is suppose to use the fake-client, therefore you should be able to fetch the VMI that was updated in the production code and see the change.

In production we have:
```go
_, err := c.clientset.VirtualMachineInstance(vmi.Namespace).Update(context.Background(), vmiCopy, v1.UpdateOptions{})
```

In the test you should be able to do this:

```go
vmi, err := virtClientset.KubevirtV1().VirtualMachineInstances(vmi.Namespace).Get(context.Background(), vmi.Name, metav1.GetOptions{})
```